Core/Core/DesignSystem/BaseScreen.swift

```diff
@@ -0,0 +1,257 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2024-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+
+public enum ScreenState: String, Equatable, Identifiable, CaseIterable {
+    case loading
+    case error
+    case empty
+    case data
+
+    public var id: String { rawValue }
+}
+
+/**
+ This structure encapsulates properties that alter the behavior of the screen.
+ Properties were collected in a mind that those won't change and are static to a specific screen.
+ */
+public struct BaseScreenConfig: Equatable {
+    public let refreshable: Bool
+    public let showsIndicators: Bool
```

Comment:
Could we change `showsIndicators` -> `showsScrollIndicators`?

Reply:
I did it, it just got lost during refactoring